### PR TITLE
Use a set in SymbolValidationManager::recordExists

### PR DESCRIPTION
**Pull request by jdmpapin**
This reduces the running time of `recordExists` from linear to log time. Furthermore, by using a separate data structure here, it's now possible to seed the set with records that should not be generated.

An ordering comparison is defined on validation records in order to allow them to be kept in a set.

### CODE REVIEW

File: runtime/compiler/runtime/SymbolValidationManager.cpp [hunk 1]
```diff
@@ -1121,6 +1121,52 @@ static void printClass(TR_OpaqueClassBlock *clazz)
...
+         LESS,
+         EQUAL,
+         GREATER,
+         };
+
+      public:
+      LexicalOrder() : _result(EQUAL) { }
+      LexicalOrder(const LexicalOrder &other) : _result(other._result) { }
+
+      static LexicalOrder by(void *a, void *b) { return LexicalOrder().thenBy(a, b); }
```
Comment (dsouzai) [inline runtime/compiler/runtime/SymbolValidationManager.cpp]: Should the return value be `LexicalOrder &`?

Comment (jdmpapin) [reply runtime/compiler/runtime/SymbolValidationManager.cpp]: There's no `LexicalOrder` we could reference here that lives past the return

Comment (dsouzai) [reply runtime/compiler/runtime/SymbolValidationManager.cpp]: I see; then I guess my question is, why have `thenBy` return a reference when passed in `void *` but return by value when passed in `uintptr_t `?

Comment (jdmpapin) [reply runtime/compiler/runtime/SymbolValidationManager.cpp]: Oh, thanks - I didn't mean to copy in `thenBy(uintptr_t, uintptr_t)`

Comment (jdmpapin) [reply runtime/compiler/runtime/SymbolValidationManager.cpp]: Fixed

**Comment by dsouzai** [comment]
Jenkins test sanity xlinux,win,plinux jdk8,jdk11

**Comment by dsouzai** [comment]
Jenkins test sanity xlinux,win,plinux jdk8,jdk11

**Comment by dsouzai** [comment]
Jenkins copyright check

**Comment by dsouzai** [comment]
Jenkins line endings check

**Pull request merged**